packages/wdio-allure-reporter/src/reporter.ts

```diff
@@ -331,65 +500,60 @@ export default class AllureReporter extends WDIOReporter {
             return
         }
 
-        this._allure.startStep(command.method
-            ? `${command.method} ${command.endpoint}`
-            : command.command
-        )
+        this._startStep(command.method ? `${command.method} ${command.endpoint}` : command.command as string)
 
         const payload = command.body || command.params
+
         if (!isEmpty(payload)) {
-            this.dumpJSON('Request', payload)
+            this.attachJSON('Request', payload)
         }
     }
 
     onAfterCommand(command: AfterCommandArgs) {
         const { disableWebdriverStepsReporting, disableWebdriverScreenshotsReporting } = this._options
-        if (this.isScreenshotCommand(command) && command.result.value) {
-            if (!disableWebdriverScreenshotsReporting) {
-                this._lastScreenshot = command.result.value
-            }
-        }
 
-        if (!this.isAnyTestRunning()) {
+        if (disableWebdriverStepsReporting || !this.currentAllureSpec || this._isMultiremote) {
```

Comment:
The condition is incorrect. If disableWebdriverStepsReporting is set to false, then disableWebdriverScreenshotsReporting is not considered at all, and no screenshots will be included in the report.